Increase alignment for bit vectors 64 => 256 bytes

baldersheim opened the pull request:
@toregge PR

baldersheim committed:
Increase alignment for bit vectors 64 => 256 bytes

toregge commented on searchlib/src/vespa/searchlib/common/bitvector.h:
```diff
@@ -295,7 +295,7 @@ class BitVector : protected BitWord
     static Index numBytes(Index bits) noexcept { return numWords(bits) * sizeof(Word); }
     size_t numWords() const noexcept { return numWords(size()); }
-    static constexpr size_t getAlignment() noexcept { return 0x40u; }
+    static constexpr size_t getAlignment() noexcept { return 0x100u; }
```
Unfortunately, https://github.com/vespa-engine/vespa/blob/c4e33003e5ce3f385951c107714ede8556ef8083/searchlib/src/vespa/searchlib/diskindex/bitvectordictionary.cpp#L61
indicates that we currently don't store the aligned bitvector byte size  in the file header, thus old bitvector files will be misread when changing the alignment. We likely need a getLegacyFileBytes() method that returns old vector size based on old alignment, used when bit the vector file lacks a file header tag for aligned bitvector byte size.
We should add a file header tag for new bitvector files, and use that value if present instead of the implicit legacy value.

baldersheim replied:
Solved as suggested.

baldersheim committed:
Pad to required alignment.

baldersheim committed:
- Store entrySize in fileheader and use that as entry size if present. If not use legacy alignment and compute size.
- Make predefined tags for use in fileheaders to reduce string constants lying around.

baldersheim commented:
@toregge PTAL

toregge commented on searchlib/src/vespa/searchlib/common/fileheadertags.cpp:
```diff
@@ -0,0 +1,16 @@
+vespalib::string FREEZE_TIME("freezeTime");
+vespalib::string CREATE_TIME("createTime");
+vespalib::string FROZEN("frozen");
+vespalib::string DOCID_LIMIT("docidLimit");
```
Should be "docIdLimit"

baldersheim replied:
#29848
